src/proof/arith_proof.cpp

```diff
@@ -870,6 +870,56 @@ void LFSCArithProof::printOwnedSort(Type type, std::ostream& os) {
   }
 }
 
+std::string LFSCArithProof::getLfscFunction(const Node & n) {
+  Assert(n.getType().isInteger() || n.getType().isReal() || n.getType().isBoolean());
+  std::string opString;
+  switch (n.getKind()) {
+    case kind::UMINUS:
+      opString = "u-_";
+      break;
+    case kind::PLUS:
+      opString = "+_";
+      break;
+    case kind::MINUS:
+      opString = "-_";
+      break;
+    case kind::MULT:
+      opString = "*_";
+      break;
+    case kind::DIVISION:
+    case kind::DIVISION_TOTAL:
+      opString = "/_";
+      break;
+    case kind::GT:
+      opString = ">_";
+      break;
+    case kind::GEQ:
+      opString = ">=_";
+      break;
+    case kind::LT:
+      opString = "<_";
+      break;
+    case kind::LEQ:
+      opString = "<=_";
+      break;
+    default:
+      Unreachable() << "Tried to get the operator for a non-operator kind: " << n.getKind();
+  }
+  std::string typeString;
+  if (n.getType().isInteger()) {
+    typeString = "Int";
+  } else if (n.getType().isReal()) {
+    typeString = "Real";
+  } else { // Boolean
+    if (n[0].getType().isInteger()) {
+      typeString = "IntReal";
```

Comment:
Can't this ever be "Int"? Like in ">=_Int"?

Reply:
Actually no. If the constant on the right is an integer, we still lift it to a Real.

Reply:
Really good question though. The refinement that you suggest would be sound, but it doesn't allow any new reasoning to be applied.

Reply:
How do you know that it is a constant? Maybe assert it?

Reply:
Good idea. Do you think that assertion would be better located here, or in `printOwnedTermAsType` (which will call this soon)?

Reply:
Personally I would probably put it in both in fear of future mis-use of both (my suggestion is an assertion that will run only on debug, not on production). But this might be an overkill.  Completely up to you.

Reply:
Sounds good. I feel weird about putting it here because there's nothing about getting a predicates name that implies the right hand side is constant. The constant-ness is pretty unrelated. What I'll do it document the fact that `Int` predicates are never returned.